index.bs

```diff
@@ -212,40 +215,29 @@ When asked to <dfn abstract-op>append client hints to request</dfn> with |settin
 |hintSet|.
- <li>For each |possibleHint| in |hintSet|, if |request| is a [=subresource request=] and the result of
-running [[permissions-policy#algo-should-request-be-allowed-to-use-feature]] given |request| and
-|possibleHint|'s associated feature in [[#policy-controlled-features]] returns `false`,
-[=list/remove=] |possibleHint| from |hintSet|.
- <li>For each |hintName| in |hintSet|, if the |request|'s [=request/header list=] [=header list/does not
-contain=] |hintName|:
+ <li>For each |hintName| in |hintSet|:
  <ol>
-  <li>Let |value| be the result of running [=find client hint value=] with |hintName|.
-  <li>[=header list/append=] |hintName|/|value| to the [=request/header list=]
+   <li>If |request| is not a [=navigation request=] for a "document" [=request/destination=] and if the result of running [[permissions-policy#algo-should-request-be-allowed-to-use-feature]]
+   given |request| and |hintName|'s associated feature in [[#policy-controlled-features]] returns `false`, then continue to next |hintName|.
+   <li>Let |value| be the result of running [=find client hint value=] with |hintName|.
+   <li>[=header list/append=] |hintName|/|value| to the [=request/header list=].
  </ol>
 </ol>
 
-When asked to <dfn abstract-op>update client hints from redirect if needed</dfn> with |request| as input, run the following steps:
+When asked to <dfn abstract-op>remove client hints from redirect if needed</dfn> with |request| as input, run the following steps:
 
 <ol>
- <li>If |request|'s [=request/client=] is null, then abort these steps.
- <li>Let |clientHintsSet| be |request|'s <var ignore>client</var>'s [=environment settings object/client hints set=].
- <li><p><a for=list>For each</a> <var>hintName</var> of |clientHintsSet|:
+ <li><p><a for=list>For each</a> <var>hintToken</var> in the list of [=client hints token=]s:
  <ol>
-  <li><p>Set <var>hintName</var> to "Sec-" concatenated with <var>hintName</var>.
   <li><p>If <var>request</var>'s <a for=request>header list</a> <a for="header list">contains</a>
-  <var>hintName</var> and if the result of running [=Should request be allowed to use feature?=],
-  given <var>request</var> and <var>hintName</var>’s <a href="#policy-controlled-features">associated
-  policy-controlled feature</a>, returns <code>false</code>, then remove <var>hintName</var> from |request|'s [=request/header list=].
-  <li><p>Else if <var>request</var>'s <a for=request>header list</a> doesn't <a for="header list">contain</a>
-  <var>hintName</var> and if the result of running [=Should request be allowed to use feature?=],
-  given <var>request</var> and <var>hintName</var>’s <a href="#policy-controlled-features">associated
-  policy-controlled feature</a>, returns <code>true</code>, then add <var>hintName</var> to |request|'s [=request/header list=].
+  <var>hintName</var>, then remove <var>hintName</var> from |request|'s [=request/header list=].
```

Comment:
Why are we removing all the hints from redirects? Shouldn't they take permission policy into account?

Reply:
We're removing them before they get re-added by the main fetch algorithm. It looks like fetch was restructured and now shares a common algorithm for both paths, making our spec less duplicative.

Reply:
OK, sounds great!!